airflow/utils/db.py

```diff
@@ -1143,7 +1144,7 @@ def check_run_id_null(session: Session) -> Iterable[str]:
         dagrun_table.c.run_id.is_(None),
         dagrun_table.c.execution_date.is_(None),
     )
-    invalid_dagrun_count = session.query(func.count(dagrun_table.c.id)).filter(invalid_dagrun_filter).scalar()
+    invalid_dagrun_count = session.scalar(select(func.count(dagrun_table.c.id)).filter(invalid_dagrun_filter))
```

Comment:
```suggestion
    invalid_dagrun_count = session.scalar(select(func.count(dagrun_table.c.id)).where(invalid_dagrun_filter))
```
I think `filter` is removed in 2.0, also, it feels much better when you read the query with `where`

Reply:
fixed it

Reply:
please check [c56bafc](https://github.com/apache/airflow/pull/32177/commits/c56bafc4f4ef7f68819b4901081cb2ad1112e461)